### PR TITLE
Update docs for 1.0.0

### PR DESCRIPTION
**Pull request by netdpb**

### CODE REVIEW

File: docs/docs/whether.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+sidebar_position: 3
+---
+
+# Should you use JSpecify annotations in your Java code?
+
+With the release of JSpecify 1.0, we guarantee backwards compatibility: we will
```
Comment (wmdietl) [inline docs/docs/whether.md:7]:
```suggestion
With the release of JSpecify 1.0.0, we guarantee backwards compatibility: we will
```

I've seen that this was done in other places...

**Review by wmdietl — APPROVED** [review]
Looks great, just three small comments to consider.

**Comment by cpovirk** [comment]
In hindsight, maybe I should have done all that as one commit. But I didn't see the button for it at first. Oh, well.

**Comment by netdpb** [comment]
I will likely use `git` command line to rebase `docs-1.0` onto `main`. I can squash those individual commits.

**Comment by netdpb** [comment]
This was done via #573 .

**Pull request closed without merging**